cluster-api/pkg/controller/machineset/utils.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machineset
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	machinelistersv1alpha1 "k8s.io/kube-deploy/cluster-api/client/listers/cluster/v1alpha1"
+)
+
+const (
+	machineCacheCheckInterval = 100 * time.Millisecond
+	machineCacheCheckTimeout  = 10 * time.Second
+)
+
+func waitForMachineInCache(name string, lister machinelistersv1alpha1.MachineLister) error {
+	return wait.PollImmediate(machineCacheCheckInterval, machineCacheCheckTimeout, func() (done bool, err error) {
+		if _, err := lister.Get(name); err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		return true, nil
+	})
+}
+
+func waitForMachineDeletedInCache(name string, lister machinelistersv1alpha1.MachineLister) error {
```

Comment:
Why is this needed?

Reply:
you are right, this brings more confusion than value.